frontend/src/concepts/analyticsTracking/useTrackUser.ts

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import { useUser } from '~/redux/selectors';
+import { useAccessReview } from '~/api';
+import { AccessReviewResourceAttributes } from '~/k8sTypes';
+import { IdentifyEventProperties } from '~/concepts/analyticsTracking/trackingProperties';
+
+export const useTrackUser = (username?: string): [IdentifyEventProperties, boolean] => {
+  const { isAdmin } = useUser();
+  const [anonymousId, setAnonymousId] = React.useState<string | undefined>(undefined);
+
+  const [loaded, setLoaded] = React.useState(false);
+  const createReviewResource: AccessReviewResourceAttributes = {
+    group: 'project.openshift.io',
+    resource: 'projectrequests',
+    verb: 'create',
+  };
+  const [allowCreate, acLoaded] = useAccessReview(createReviewResource);
+
+  React.useEffect(() => {
+    const computeAnonymousUserId = async () => {
+      const anonymousIDBuffer = await crypto.subtle.digest(
+        'SHA-1',
+        new TextEncoder().encode(username),
+      );
+      const anonymousIDArray = Array.from(new Uint8Array(anonymousIDBuffer));
+      const aId = anonymousIDArray.map((b) => b.toString(16).padStart(2, '0')).join('');
+      return aId;
+    };
+
+    if (!anonymousId) {
+      computeAnonymousUserId().then((val) => {
+        setAnonymousId(val);
+      });
+    }
+    if (acLoaded && anonymousId) {
+      setLoaded(true);
+    }
+  }, [username, anonymousId, acLoaded]);
```

Comment:
If `username` changes, you aren't recomputing the anonymous id. Although I don't think `username` should change and you are only going to compute the anonymous id once. But should you recompute?

If you don't care to recompute based on `username` changes, then you can remove all hook dependencies.
```suggestion
    computeAnonymousUserId().then((val) => {
      setAnonymousId(val);
    });
  // compute anonymousId only once
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
```

Reply:
The usename is set at login. So I don't think it will ever change during one session.